### DISCUSSION TITLE
Is it possible to run an existing workflow template in Hera?

### DISCUSSION THREAD

**Discussion by fel-este** [argoproj-labs/hera#1188] [Q&A] [answered]
I am looking at the examples for Hera and all generate a workflow template via Hera. I have a workflow template in YAML that doesn't need to be recreated every time by Hera. Is there a way to specify a workflow template that already exists in an Argo namespace?

**Comment by elliotgunton** [comment, marked as answer]
Hi @fel-este - to run an existing workflow template you can use a `WorkflowTemplateRef` seen [here](https://hera.readthedocs.io/en/stable/examples/workflows/upstream/workflow_template__workflow_template_ref), or a regular `TemplateRef` seen [here](https://hera.readthedocs.io/en/stable/examples/workflows/upstream/workflow_template__hello_world/)
